components/brave_shields/browser/ad_block_subscription_service_manager.cc

```diff
@@ -327,6 +366,12 @@ void AdBlockSubscriptionServiceManager::OnGetDownloadManager(
   download_manager_->CancelAllPendingDownloads();
   LoadSubscriptionServices();
 
+  if (subscriptions_.empty() && !brave::CanUseBraveServices()) {
```

Comment:
buildflag is a compile-time thing. If you can make this part to be a compile-time conditional, then it would be great.
basically we don't need to pull `const char* DEFAULT_LISTS` into the final binary if the bulidflag is set.

Reply:
hmm, looks... [impossible](https://stackoverflow.com/questions/2335888/how-to-compare-strings-in-c-conditional-preprocessor-directives)? (thanks, @bbondy :wink:)

Is there a way to have GN define another flag that's _not_ a string, just for this purpose?

Reply:
> hmm, looks... [impossible](https://stackoverflow.com/questions/2335888/how-to-compare-strings-in-c-conditional-preprocessor-directives)? (thanks, @bbondy 😉)
> 
> Is there a way to have GN define another flag that's _not_ a string, just for this purpose?


`if constexpr` [is allowed](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++-features.md#constexpr-if-allowed). this should do the job:
```
// declare this in header
constexpr bool CanUseBraveServices() {
  return !base::StringPiece(BUILDFLAG(BRAVE_SERVICES_KEY)).empty();
}

// declare this instead if DEFAULT_LISTS
base::span<const base::StringPiece> GetDefaultLists() {
  if constexpr (base::StringPiece(BUILDFLAG(BRAVE_SERVICES_KEY)).empty()) {
    static constexpr base::StringPiece kDefaultLists[] = { ... };
    return kDefaultLists;
  } else {
    return {};
  }
}
```

Reply:
Worked really well, thanks @goodov!